src/python/examples/unittests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,303 @@
+        server_url = sqlalchemy.engine.url.make_url(request.config.getoption('server'))
+        type(self).host = server_url.host
+        type(self).port = server_url.port
+        type(self).username = "ensro" if server_url.username == "ensadmin" else server_url.username
+
+    @pytest.mark.skipif(os.environ['USER'] == 'travis',
+                        reason="The test requires both Perl and Python which is not supported by Travis.")
+    @pytest.mark.parametrize(
+        "core_list, species_set_name, id_type, expectation",
+        [
+            (
+                [f"{os.environ['USER']}_gallus_gallus_core_99_6",
+                 f"{os.environ['USER']}_homo_sapiens_core_99_38"],
+                "default", "protein", does_not_raise()
+            ),
+            ([], "test", "gene", raises(ValueError, match=r"No cores to dump."))
+        ]
+    )
+    def test_dump_genomes(self, core_list: List[str], species_set_name: str,
+                          tmp_dir: Path, id_type: str, expectation: ContextManager) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` when server connection can be established.
+
+        Args:
+            species_list: A list of species (genome names).
+            species_set_name: Species set (collection) name.
+            tmp_dir: Unit test temp directory (fixture).
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            orthology_benchmark.dump_genomes(core_list, species_set_name, self.host, self.port, tmp_dir,
+                                             id_type)
+
+            out_files = tmp_dir / species_set_name
+            # pylint: disable-next=no-member
+            exp_out = pytest.files_dir / "orth_benchmark"  # type: ignore[attr-defined]
+            for db_name, unittest_db in self.core_dbs.items():
+                assert file_cmp(out_files / f"{unittest_db.dbc.db_name}.fasta", exp_out / f"{db_name}.fasta")
+
+    def test_dump_genomes_fake_connection(self, tmp_dir: Path) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` with fake server details.
+
+        Args:
+            tmp_dir: Unit test temp directory (fixture).
+
+        """
+        with raises(RuntimeError):
+            orthology_benchmark.dump_genomes(["mus_musculus", "naja_naja"], "fake",
+                                             "fake-host", 65536, tmp_dir, "protein")
+
+    def test_dump_genomes_fake_output_path(self) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` with fake output path."""
+        with raises(OSError, match=r"Failed to create '/compara/default' directory."):
+            orthology_benchmark.dump_genomes(["mus_musculus", "naja_naja"], "default",
+                                             self.host, self.port, "/compara", "protein")
```

Comment:
As above, I'd suggest at least renaming the nonexistent path to be as explicit as possible about its purpose:
```suggestion
        with raises(OSError, match=r"Failed to create '/nonexistent/path/default' directory."):
            orthology_benchmark.dump_genomes(["mus_musculus", "naja_naja"], "default",
                                             self.host, self.port, "/nonexistent/path", "protein")
```